spec.bs

```diff
@@ -972,14 +752,350 @@ To <dfn>generate reporting URLs</dfn> given an {{AuctionAdConfig}} |config| and
+  To <dfn>obtain a script runner agent</dfn>, run these steps:
+
+    1. Let |agentCluster| be a  new [=ECMAScript/agent cluster=].
+
+    1. Let |agent| be the result of [=creating a new script runner agent=].
+
+    1. Add |agent| to |agentCluster|.
+
+    1. Return |agent|.
+</div>
+
+<div algorithm>
+  To <dfn>evaluate a script</dfn> with a global type |globalType|, [=string=] |script|, [=string=]
+  |functionName|, and a [=list=] |arguments|, run these steps. They return a [=ECMAScript/Completion
+  Record=], which is either an [=abrupt completion=] (in the case of a parse failure or execution
+  error), or a normal completion populated with the [=ECMAScript/ECMAScript language value=] result
+  of invoking |functionName|.
+
+    1. [=Assert=] that these steps are running [=in parallel=].
+
+    1. Let |agent| be the result of [=obtaining a script runner agent=] given null, true, and
+       false. Run the rest of these steps in |agent|.
+
+       Issue: This exclusively creates a new [=ECMAScript/agent cluster=] for the given |script| to
+       run in, but we should make this work with [=interest group/execution mode=] somehow.
+
+    1. Let |realmExecutionContext| be the result of [=creating a new realm=] given |agent| and the
+       following customizations:
+
+        * For the global object, create a new object of type |globalType|.
+
+    1. Let |realm| be |realmExecutionContext|'s Realm component.
+
+    1. Let |global| be |realm|'s [=realm/global object=], and run these steps:
+
+      1. Perform !|global|.\[[Delete]]("`Date`").
+
+      1. If !|global|.\[[HasProperty]]("`Temporal`") is true, then perform
+         !|global|.\[[Delete]]("`Temporal`").
+
+      Advisement: This is not the best way to perform such API neutering (see <a
+      href=https://github.com/tc39/ecma262/issues/1357#issuecomment-817560121>tc39/ecma262#1357</a>),
+      but at the moment it's the way that host environments do this.
+
+      Note: Removing time-referencing APIs from the |global| object is imperative for privacy, as
+      |script| would otherwise be able to accurately exfiltrate data gleaned from time measurements.
+
+    1. Let |result| be [=ParseScript=](|script|, |realm|, `empty`).
+
+       Note: The resulting [=ECMAScript/Script Record=] will have no \[[HostDefined]] component,
+       unlike traditional [=scripts=] on the web platform.
+
+    1. If |result| is a list of errors, return
+       Completion { \[[Type]]: `throw`, \[[Value]]: |result|, \[[Target]]: `empty` }.
+
+    1. [=Assert=]: |result| is a [=ECMAScript/Script Record=].
+
+    1. *Prepare to run script*: Push |realmExecutionContext| onto the [=ECMAScript/execution context
+       stack|JavaScript execution context stack=]; it is now the [=ECMAScript/running execution
+       context|running JavaScript execution context=].
+
+    1. Let |evaluationStatus| be the result of [=ScriptEvaluation=](result).
+
+    1. If |evaluationStatus| is an [=abrupt completion=], FOO.

```

Comment:
s/FOO/TODO/?

Reply:
Good catch, this should actually just jump to the step labeled to return, like the other steps in between here and there.